monkey/infection_monkey/network/tcp_scanner.py

```diff
@@ -58,31 +59,18 @@ def _check_tcp_ports(ip: str, ports: List[int], timeout: float = DEFAULT_TIMEOUT
                 connected_ports_sockets.append((port, sock))
                 possible_ports.append((port, sock))
                 continue
-            # BUG: I don't think a socket will ever connect successfully if this error is raised.
-            #      From the documentation: "Resource temporarily unavailable... It is a nonfatal
-            #      error, **and the operation should be retried later**." (emphasis mine). If the
-            #      operation is not retried later, I don't see the point in appending this to
-            #      possible_ports.
-            if err == 10035:  # WSAEWOULDBLOCK is valid, see
-                # https://msdn.microsoft.com/en-us/library/windows/desktop/ms740668%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396
-                possible_ports.append((port, sock))
-                continue
```

Comment:
@shreyamalviya Did you investigate this any more to find out if this is the case? The documentation is somewhat ambiguous. It says "the operation should be retried later." Does that mean the caller should retry or you shouldn't worry about it because the os should try it later?

Reply:
@shreyamalviya Looks like my initial reading of the documentation was incorrect. It's not up to the caller to retry:
![image](https://user-images.githubusercontent.com/19957806/153263108-74bd635f-49f4-406c-9701-6ae7fd50cd47.png)
https://docs.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-connect

I think we should just clarify the existing comment, add the above link, and remove my `BUG` comment, but the if check should stay based on the above documentation.